Documentation/ApiOverview/Bootstrapping/Index.rst

```diff
@@ -7,60 +7,78 @@ Bootstrapping
 
-Some contexts add their own bootstrap class (like the command
-line, which additionally requires :php:`\TYPO3\CMS\Core\Core\CliBootstrap`).
+Each application class registers Request Handlers to the TYPO3 Bootstrap to
+run a certain request type (e.g. eID or TSFE-logic, or AJAX requests in the Backend). Each application is handed
+over the class loader provided by Composer.
 
-.. note::
+Applications
+============
 
-   The frontend's bootstrapping process is not yet fully encapsulated
-   in a bootstrap class.
+There are four types of applications provided by the TYPO3 Core:
 
-.. warning::
 
-   This bootstrapping API is internal and may change at any time in the near future
-   even in minor updates. It is thus discouraged to use it in third party code.
-   Use this class only if other extensibility possibilities such as
-   :ref:`Hooks <hooks>`, Signals or :ref:`XCLASS <xclasses>`
-   are not enough to reach your goals.
+\\TYPO3\\CMS\\Frontend\\Http\\Application
+-----------------------------------------
 
+This class handles all incoming web requests coming through :file:`index.php`
+in the main directory. It handles all TSFE and eID requests.
 
-One can see the bootstrapping process in action in file
-:file:`typo3/sysext/backend/Classes/Http/Application.php`::
+It checks if all configuration is set, otherwise redirects to the TYPO3 Install
+Tool.
 
-   use TYPO3\CMS\Core\Core\Bootstrap;
 
-   ###
+\\TYPO3\\CMS\\Backend\\Http\\Application
+----------------------------------------
 
-   $this->bootstrap = Bootstrap::getInstance()
-      ->initializeClassLoader($classLoader)
-      ->setRequestType(TYPO3_REQUESTTYPE_BE | (!empty($_GET['ajaxID']) ? TYPO3_REQUESTTYPE_AJAX : 0))
-      ->baseSetup($this->entryPointLevel);
+This class handles all incoming web requests for any regular backend call
+inside :file:`typo3/\*`.
 
-   // Redirect to Install Tool if base configuration is not found
-   if (!$this->bootstrap->checkIfEssentialConfigurationExists()) {
-      $this->bootstrap->redirectToInstallTool($this->entryPointLevel);
-   }
+Its :php:`TYPO3\CMS\Backend\Http\RequestHandler` is used for all Backend
+requests, including AJAX routes. If a get/post parameter "route" is set, the
+Backend Routing is called by the :php:`RequestHandler` and
+searches for a matching route inside the Router. The corresponding controller
+/ action is called then which returns the response.
 
-   foreach ($this->availableRequestHandlers as $requestHandler) {
-      $this->bootstrap->registerRequestHandlerImplementation($requestHandler);
-   }
+The :php:`Application` checks if all configuration is set, otherwise it
+redirects to the TYPO3 Install Tool.
 
-   $this->bootstrap->configure();
 
-   ###
+\\TYPO3\\CMS\\Core\\Console\\CommandApplication
+-----------------------------------------------
+
+This class is the entry point for the TYPO3 command line for console commands.
+In addition to a simple Symfony command, this also sets up a CLI user.
```

Comment:
What is the "simple Symfony command"?

Reply:
No idea, do you know? Or should we just remove the sentence? (Please note I only try to get this old PR merged, I didn't write the texts in the first place)